app/models/nic/interface.rb

```diff
@@ -105,12 +105,12 @@ def normalize_name
       # no hostname was given or a domain was selected, since this is before validation we need to ignore
       # it and let the validations to produce an error
       return if name.empty?
-      if domain.nil? && name.include?('.') && changed_attributes['domain_id'].blank?
+      if domain_id.nil? && name.include?('.') && changed_attributes['domain_id'].blank?
         # try to assign the domain automatically based on our existing domains from the host FQDN
-        self.domain = Domain.find_by(:name => name.partition('.')[2])
+        self.domain = Domain.unscoped.find_by(:name => name.partition('.')[2])
```

Comment:
This and the line below with `unscoped` be able to set domains that are **not** visible to the user? E.g: I create an interface through the API with a domain name that I know I can't access. This line would find it and set it. Is this required to make `subman` work? I think the change in the  `if` condition should be enough.

Reply:
wouldn't the controller handle that?

Reply:
oh right, here its based off name, duh ;)

Reply:
the one below i would think would be filtered by the controller, or is that not how it works?

Reply:
The controller does not filter the parameters, I can send the `domain_id` of a domain I can't see (but still exists, maybe an admin can see it) as a parameter, and the domain will just pass it on. It's the model's responsibility to be unable to find the `domain_id`, then the controller would just translate that to 404.

Reply:
I would think so much would be broken as far as auth goes if that were true since update_attributes would be relied on,  but i'll take your word for it :)